in_toto/models/metadata.py

```diff
@@ -180,76 +180,72 @@ def sign_gpg(self, gpg_keyid=None, gpg_home=None):
-      keys_dict:
-              Verifying keys in the format:
-              in_toto.formats.ANY_VERIFY_KEY_DICT_SCHEMA
+      verification_key:
+              Verifying key in the format:
+              in_toto.formats.ANY_VERIFICATION_KEY_SCHEMA
 
     <Exceptions>
       FormatError
-            if the passed key dictionary is not conformant with
-            in_toto.formats.ANY_VERIFY_KEY_DICT_SCHEMA
+            If the passed key is not conformant with
+            `in_toto.formats.ANY_VERIFICATION_KEY_SCHEMA`
 
       SignatureVerificationError
-            if the Metablock is not signed
+            If the Metablock does not carry a signature signed with the
+            private key corresponding to the passed verification key
 
-            if the Metablock carries a signature for which no key is found in
-            the passed key dictionary, which means that multiple signatures
-            have to be verified at once
+            If the signature corresponding to the passed verification key
+            does not match securesystemslib's or in_toto.gpg's
+            signature schema.
 
-            if any of the verified signatures is actually broken
+            If the verified signature is invalid.
 
     <Returns>
       None.
 
     """
-    in_toto.formats.ANY_VERIFY_KEY_DICT_SCHEMA.check_match(keys_dict)
-
-    if not self.signatures or len(self.signatures) <= 0:
-      raise SignatureVerificationError("No signatures found")
+    in_toto.formats.ANY_VERIFICATION_KEY_SCHEMA.check_match(verification_key)
+    verification_keyid = verification_key["keyid"]
 
+    # Find a signature that corresponds to this verification keyid and raise
+    # an exception if it doesn't exist
+    signature = None
     for signature in self.signatures:
-      keyid = signature["keyid"]
-      try:
-        key = keys_dict[keyid]
+      if signature["keyid"] == verification_keyid:
+        break
+    else:
+      raise SignatureVerificationError("No signature found for key '{}'"
+          .format(verification_keyid))
 
-      except KeyError:
-        raise SignatureVerificationError(
-            "Signature key not found, key id is '{0}'".format(keyid))
+    if in_toto.gpg.formats.SIGNATURE_SCHEMA.matches(signature):
+      valid = in_toto.gpg.functions.gpg_verify_signature(signature,
+          verification_key, self.signed.signable_bytes)
 
-      if in_toto.gpg.formats.SIGNATURE_SCHEMA.matches(signature):
-        valid = in_toto.gpg.functions.gpg_verify_signature(signature, key,
-            self.signed.signable_bytes)
+    elif securesystemslib.formats.SIGNATURE_SCHEMA.matches(signature):
+      valid = securesystemslib.keys.verify_signature(
+          verification_key, signature, self.signed.signable_dict)
 
-      else:
-        valid = securesystemslib.keys.verify_signature(
-            key, signature, self.signed.signable_dict)
+    else:
+      # TODO: Add Metablock format validator that prevent malformed signatures
+      valid = False
```

Comment:
I think that an error should be raised here indicating that the key is not in a known or supported format.

Reply:
I thought about that too, but I think it's okay (for now) to treat a malformed signature and an invalid signature alike. In any case, there is a TODO comment, to move the format validation elsewhere.